plasmapy/dispersion/numerical/hirose_.py

```diff
@@ -0,0 +1,316 @@
+"""
+This module contains functionality for calculating various numerical
+solutions to Hirose's two fluid dispersion relation, see
+:cite:t:`hirose:2004` and :cite:t:`bellan:2012`.
+"""
+
+import astropy.units as u
+import numpy as np
+import warnings
+
+from astropy.constants.si import c
+from typing import Optional, Union
+
+from plasmapy.formulary.frequencies import gyrofrequency, plasma_frequency
+from plasmapy.formulary.speeds import Alfven_speed, ion_sound_speed
+from plasmapy.particles import Particle
+from plasmapy.particles.exceptions import ChargeError
+from plasmapy.utils.decorators import validate_quantities
+from plasmapy.utils.exceptions import PhysicsWarning
+
+c_si_unitless = c.value
+
+
+@validate_quantities(
+    B={"can_be_negative": False},
+    n_i={"can_be_negative": False},
+    T_e={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+)
+def hirose(
+    *,
+    B: u.T,
+    ion: Union[str, Particle],
+    k: u.rad / u.m,
+    n_i: u.m**-3,
+    T_e: u.K,
+    theta: u.rad,
+    gamma_e: Optional[float, int] = 1,
+    gamma_i: Optional[float, int] = 3,
+    z_mean: Optional[float, int] = None,
```

Comment:
```suggestion
    gamma_e: Optional[Union[float, int]] = 1,
    gamma_i: Optional[Union[float, int]] = 3,
    z_mean: Optional[Union[float, int]] = None,
```

This should fix the current build errors.  Sorry, the was my fault with the previous suggestion.